### ISSUE TITLE
Pylint doesn't run in parallel if -j=0 is specified

### ISSUE THREAD

**Issue by albertino87** [pylint-dev/pylint#9537]
### Bug description

```python
Hi,
when i run pylint specifying -j=2 or -j=4 and so on it works as expected, however when I run it with -j=0 it actually spawns only 1 process even though I have 8 cores. I read the code and this happen because

    cpu_share = _query_cpu()

which returns 1, so later it returns:

    if cpu_share is not None:
        return min(cpu_share, cpu_count)
```

### Command used

```shell
pylint . -j=0
```

### Pylint output

```python
normal output
```

### Expected behavior

i would expect to use all the cores of my machine

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.3
Python 3.11.6 | packaged by conda-forge |
```

### OS / Environment

Rhel 8

**Comment by Pierre-Sassoulas** [comment]
We have a hard time getting the proper number of core on all target systems and platforms. Could you provide a python snippet that return the proper number of core on rhel 8 ? (See #6965, #7035, same issue with windows)

**Comment by albertino87** [comment]
well pytest somehow gets it right

**Comment by jacobtylerwalls** [comment]
Yeah I think #7338 is the duplicate. My understanding is the last time we touched this we overfitted it to AWS and broke everything else. Thanks for the report.

**Comment by Pierre-Sassoulas** [comment]
@albertino87 how are you parallelizing pytest ? I don't think pytest permit to launch in parallel, do you mean ``pytest-xdist`` ?

**Comment by albertino87** [comment]
@Pierre-Sassoulas yes, sorry, i meant pytest-xdist
